### PR TITLE
Support dynamic has_many and has_many_ordered

### PR DESCRIPTION
**Pull request by joshuaclayton**
- Simplify selector generation by calling proc within defined methods
- Check provided selector_type to determine dynamic behavior upon name registration
- Allow use of methods for has_many and has_many_ordered

### CODE REVIEW

File: lib/page_ez/method_generators/has_many_ordered_selector.rb [hunk 1]
```diff
@@ -26,27 +25,31 @@ def run(target)
...
-        evaluator_class = SelectorEvaluator.build(@name, dynamic_options: dynamic_options, options: options, selector: build_selector)
+        evaluator_class = SelectorEvaluator.build(@name, dynamic_options: dynamic_options, options: options, selector: selector)
 
         target.logged_define_method("#{singularized_name}_at") do |index, *args|
-          evaluator = evaluator_class.run([index].dup.push(*args), target: self)
+          evaluator = evaluator_class.run(args, target: self)
           HasOneResult.new(
             container: container,
-            selector: evaluator.selector,
+            selector: build_selector[evaluator.selector, index],
```
Comment (eebs) [inline lib/page_ez/method_generators/has_many_ordered_selector.rb:34]: Nice change, it's great to be able to keep the index out of the evaluator class arguments.

File: lib/page_ez/visitors/registered_name_visitor.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def track_method_added(name, construction_strategy)
 
         found = @declared_constructors[parent_id].find { _1 == [name, construction_strategy] }
 
-        if found && !found[1].is_a?(PageEz::MethodGenerators::HasOneDynamicSelector)
+        if found && found[1]&.selector_type != :dynamic
```
Comment (eebs) [inline lib/page_ez/visitors/registered_name_visitor.rb:35]: I'm curious why you chose to use `!= :dynamic` vs `== :static`? Is the intent "anything that's not dynamic should raise" vs "only raise for static selectors"?

File: spec/features/macros_with_methods_spec.rb [hunk 3]
```diff
@@ -85,6 +85,88 @@ def section_with_id(id:)
...
+
+      def items(state:)
+        "li[data-state='#{state}']"
+      end
+    end.new(page)
+
+    page.visit "/"
+
+    expect(test_page.items(state: "complete")).to have_count_of(1)
+    expect(test_page.items(state: "incomplete")).to have_count_of(2)
```
Comment (eebs) [inline spec/features/macros_with_methods_spec.rb:120]: This reads so well ❤️

**Review by eebs — APPROVED** [review]
This is awesome, and nice to see how the previous set of changes made this relatively straightforward.

File: lib/page_ez/visitors/registered_name_visitor.rb (hunk 2 above)
Comment (joshuaclayton) [reply lib/page_ez/visitors/registered_name_visitor.rb:35]: @eebs you're spot on re: anything that's not dynamic should raise

**Pull request merged**